core/src/main/java/org/stellar/anchor/client/DefaultClientService.java

```diff
@@ -25,90 +19,83 @@
 @NoArgsConstructor
 public class DefaultClientService implements ClientService {
   static final Gson gson = GsonUtils.getInstance();
-  List<CustodialClientConfig> custodialClients = emptyList();
-  List<NonCustodialClientConfig> nonCustodialClients = emptyList();
+  List<CustodialClient> custodialClients = new ArrayList<>();
+  List<NonCustodialClient> nonCustodialClients = new ArrayList<>();
 
   /**
    * Creates a DefaultClientService instance based on the provided ClientsConfig.
    *
    * @param clientsConfig the configuration object containing client setup details
    * @return a DefaultClientService instance configured according to the provided ClientsConfig
-   * @throws InvalidConfigException if the provided ClientsConfig's file type is not supported or if
-   *     the provided config content is invalid
    */
-  public static DefaultClientService fromClientsConfig(ClientsConfig clientsConfig)
-      throws InvalidConfigException {
-    DefaultClientService clientService = new DefaultClientService();
-    if (clientsConfig.getType().equals(CLIENTS_CONFIG_TYPE_FILE)) {
-      Map<String, Object> map = parseFileToMap(clientsConfig.getValue());
-      clientService.custodialClients = getCustodialClientsFromMap(map);
-      clientService.nonCustodialClients = getNonCustodialClientsFromMap(map);
-    } else {
-      clientService.custodialClients = clientsConfig.getCustodial();
-      clientService.nonCustodialClients = clientsConfig.getNoncustodial();
-    }
-    return clientService;
+  public static DefaultClientService fromClientsConfig(ClientsConfig clientsConfig) {
+    return createDCSFromItemsList(clientsConfig.getItems());
   }
 
-  public static Map<String, Object> parseFileToMap(String filePath) throws InvalidConfigException {
+  public static List<TempClient> parseFileToList(String filePath) throws InvalidConfigException {
     try {
       String fileContent = FileUtil.read(Path.of(filePath));
       String fileExtension = FilenameUtils.getExtension(filePath).toLowerCase();
-      Map<String, Object> map;
+      Map<String, List<Object>> map;
       if ("yaml".equals(fileExtension) || "yml".equals(fileExtension)) {
```

Comment:
nit: To be consistent with `enum ClientConfigType`, can we use `enum` for the supported file extensions.